### PR TITLE
BREAKING CHANGE:(DB/Creature) Remove modelId field in creature table

### PR DESCRIPTION
**Pull request by malcrom**
## Changes Proposed:
-  Clearing model id field will allow random model to be loaded for creatures.

## Issues Addressed:
- Closes 

## SOURCE:
me

## Tests Performed:
- Been testing in game and cannot find any issues.

## How to Test the Changes:
1. Play on server

### CODE REVIEW

**Comment by Kitzunu** [comment]
Just drop the field?

**Comment by IntelligentQuantum** [comment]
> Just drop the field?

No,sometimes i use this column for spawn Teleport NPC with different model id.

**Comment by Kitzunu** [comment]
We need to deprecate it sooner or later anyways. The field has no use for blizzlike use and is in the way of implementing dual entry per spawn.

Creature_template handles models

**Comment by IntelligentQuantum** [comment]
> We need to deprecate it sooner or later anyways. The field has no use for blizzlike use and is in the way of implementing dual entry per spawn.
> 
> Creature_template handles models

I know, Many people use this source for fun or pvp servers (Like Me), and i dont want to create 20 npc with new entry just for model id

**Comment by Kitzunu** [comment]
But we cant keep things that will stop the blizzlike development.

**Comment by malcrom** [comment]
Anything this field is doing can be done is script.

**Review by acidmanifesto — APPROVED** [review]
Will mark as waiting for testing

**Comment by acidmanifesto** [comment]
What we can also do, if model ids of azerothcore  catelogue npc's would be effect. a list of which ones it is will need to be either adjusted to have a alternate model id to support the randomness (just reuse the same model id if need be), or given a reapply sql or sorts, custom modules load up last dont they?

**Comment by malcrom** [comment]
What needs to be tested?

**Review by Kitzunu — APPROVED** [review]

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -2028,25 +2028,24 @@ void ObjectMgr::LoadCreatures()
...
+        data.curmana            = fields[12].GetUInt32();
+        data.movementType       = fields[13].GetUInt8();
+        data.spawnMask          = fields[14].GetUInt8();
+        data.phaseMask          = fields[15].GetUInt32();
+        int16 gameEvent         = fields[16].GetInt8();
+        uint32 PoolId           = fields[17].GetUInt32();
+        data.npcflag            = fields[18].GetUInt32();
+        data.unit_flags         = fields[19].GetUInt32();
+        data.dynamicflags       = fields[20].GetUInt32();
         data.ScriptId           = GetScriptId(fields[22].GetString());
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.cpp:2049]:
```suggestion
        data.ScriptId           = GetScriptId(fields[21].GetString());
```

**Comment by FrancescoBorzi** [comment]
> > We need to deprecate it sooner or later anyways. The field has no use for blizzlike use and is in the way of implementing dual entry per spawn.
> > Creature_template handles models
> 
> I know, Many people use this source for fun or pvp servers (Like Me), and i dont want to create 20 npc with new entry just for model id

what if we keep this column and set all rows to `0` in a way that the core does like:

- if `modelid=0` then do like this field didn't exist
- else use it (useful for custom purposes)

**Comment by Kitzunu** [comment]
no it hinders https://github.com/azerothcore/azerothcore-wotlk/pull/10115

for custom stuff, just use sai or script

**Comment by FrancescoBorzi** [comment]
also I don't get how simply removing it would make the core work as intended, can anyone please clarify? just for my understanding

**Comment by FrancescoBorzi** [comment]
> no it hinders #10115
> 
> for custom stuff, just use sai or script

fine for me then, but can anyone please explain why? I'd like to understand it

**Comment by FrancescoBorzi** [comment]
Ok this has been clarified to me on Discord. I'm fine with this change, just please:

- add it to the changelog (see https://github.com/azerothcore/wiki/blob/master/docs/how-to-use-changelog.md ), including how to deal with custom model IDs from now on (e.g. using SAI)
- update the wiki accordingly
- open a ticket on Keira3 so we don't forget to update that too
- commit this starting with `BREAKING CHANGE(....): bla bla` as per https://www.conventionalcommits.org/

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
```sql
UPDATE `creature_template` SET `modelid1`=6762,`modelid2`=14886,`modelid3`=23832,`modelid4`=23833 WHERE `entry` = 26477;
UPDATE `creature_template` SET `modelid1`=20737,`modelid2`=0 WHERE `entry` = 30058;
UPDATE `creature_template` SET `modelid1`=25850,`modelid2`=25851 WHERE `entry` = 29019;
UPDATE `creature_template` SET `modelid1`=25734,`modelid2`=25735 WHERE `entry` = 28852;
UPDATE `creature_template` SET `modelid1`=25023,`modelid2`=25024,`modelid3`=25025,`modelid4`=25077 WHERE `entry` = 28260;
UPDATE `creature_template` SET `modelid1`=23246,`modelid2`=23247,`modelid3`=23248,`modelid4`=23249 WHERE `entry` = 25342;
UPDATE `creature_template` SET `modelid1`=23124,`modelid2`=23125,`modelid3`=23126,`modelid4`=0 WHERE `entry` = 25343;
UPDATE `creature_template` SET `modelid1`=22979,`modelid2`=22980 WHERE `entry` = 24918;
UPDATE `creature_template` SET `modelid1`=22911,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 24938;
UPDATE `creature_template` SET `modelid1`=22753,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 25115;
UPDATE `creature_template` SET `modelid1`=21769,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 7951;
UPDATE `creature_template` SET `modelid1`=15744,`modelid2`=15745,`modelid3`=15746,`modelid4`=15747 WHERE `entry` = 15760;

```

**Comment by acidmanifesto** [comment]
> ```sql
> UPDATE `creature_template` SET `modelid1`=6762,`modelid2`=14886,`modelid3`=23832,`modelid4`=23833 WHERE `entry` = 26477;
> UPDATE `creature_template` SET `modelid1`=20737,`modelid2`=0 WHERE `entry` = 30058;
> UPDATE `creature_template` SET `modelid1`=25850,`modelid2`=25851 WHERE `entry` = 29019;
> UPDATE `creature_template` SET `modelid1`=25734,`modelid2`=25735 WHERE `entry` = 28852;
> UPDATE `creature_template` SET `modelid1`=25023,`modelid2`=25024,`modelid3`=25025,`modelid4`=25077 WHERE `entry` = 28260;
> UPDATE `creature_template` SET `modelid1`=23246,`modelid2`=23247,`modelid3`=23248,`modelid4`=23249 WHERE `entry` = 25342;
> UPDATE `creature_template` SET `modelid1`=23124,`modelid2`=23125,`modelid3`=23126,`modelid4`=0 WHERE `entry` = 25343;
> UPDATE `creature_template` SET `modelid1`=22979,`modelid2`=22980 WHERE `entry` = 24918;
> UPDATE `creature_template` SET `modelid1`=22911,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 24938;
> UPDATE `creature_template` SET `modelid1`=22753,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 25115;
> UPDATE `creature_template` SET `modelid1`=21769,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 7951;
> UPDATE `creature_template` SET `modelid1`=15744,`modelid2`=15745,`modelid3`=15746,`modelid4`=15747 WHERE `entry` = 15760;
> ```

can you provide some context with your sql?

**Comment by claudiodfc** [comment]
> > ```sql
> > UPDATE `creature_template` SET `modelid1`=6762,`modelid2`=14886,`modelid3`=23832,`modelid4`=23833 WHERE `entry` = 26477;
> > UPDATE `creature_template` SET `modelid1`=20737,`modelid2`=0 WHERE `entry` = 30058;
> > UPDATE `creature_template` SET `modelid1`=25850,`modelid2`=25851 WHERE `entry` = 29019;
> > UPDATE `creature_template` SET `modelid1`=25734,`modelid2`=25735 WHERE `entry` = 28852;
> > UPDATE `creature_template` SET `modelid1`=25023,`modelid2`=25024,`modelid3`=25025,`modelid4`=25077 WHERE `entry` = 28260;
> > UPDATE `creature_template` SET `modelid1`=23246,`modelid2`=23247,`modelid3`=23248,`modelid4`=23249 WHERE `entry` = 25342;
> > UPDATE `creature_template` SET `modelid1`=23124,`modelid2`=23125,`modelid3`=23126,`modelid4`=0 WHERE `entry` = 25343;
> > UPDATE `creature_template` SET `modelid1`=22979,`modelid2`=22980 WHERE `entry` = 24918;
> > UPDATE `creature_template` SET `modelid1`=22911,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 24938;
> > UPDATE `creature_template` SET `modelid1`=22753,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 25115;
> > UPDATE `creature_template` SET `modelid1`=21769,`modelid2`=0,`modelid3`=0,`modelid4`=0 WHERE `entry` = 7951;
> > UPDATE `creature_template` SET `modelid1`=15744,`modelid2`=15745,`modelid3`=15746,`modelid4`=15747 WHERE `entry` = 15760;
> > ```
> 
> can you provide some context with your sql?

Tested

**Comment by acidmanifesto** [comment]
@claudiodfc not exactly what i ment. modelids are wdb fields. They are obtained by sniffs only.

**Comment by claudiodfc** [comment]
Sorry acid, I wanted to make an old joke, forget about it

**Comment by Nyeriah** [comment]
the kernel is all spoiled

**Comment by pklloveyou** [comment]
Better than the wrong model

**Pull request merged**